### PR TITLE
Akka singleton setup

### PR DESCRIPTION
**Pull request by bhoggard**

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -11,4 +11,8 @@ akka {
       local.dir = "target/snapshots"
     }
   }
-}
\ No newline at end of file
+
+  actor {
+    provider = "akka.cluster.ClusterActorRefProvider"
```
Comment (acjay) [inline src/main/resources/reference.conf:16]: Nice, good to see we're minimizing what's going in to `reference.conf`

File: src/main/scala/net/artsy/atomic/AtomicEventStore.scala [hunk 2]
```diff
@@ -127,6 +127,11 @@ abstract class AtomicEventStore[EventType <: Serializable: Scoped: ClassTag, Val
   }
 
   /**
+   * Message replied to sender when a log for a given scope is terminated.
```
Comment (acjay) [inline src/main/scala/net/artsy/atomic/AtomicEventStore.scala]: Would be more accurate to say that this message is broadcast.

File: src/main/scala/net/artsy/atomic/AtomicEventStore.scala [hunk 3]
```diff
@@ -181,35 +186,66 @@ abstract class AtomicEventStore[EventType <: Serializable: Scoped: ClassTag, Val
...
         case None =>
-          // Recreate the log, which will recall all preexisting events
-          val materializedLog = context.actorOf(logProps(scope), scope)
+          val system = context.system
+          val singletonProps = ClusterSingletonManager.props(
+            singletonProps     = logProps(scope),
+            terminationMessage = PoisonPill,
+            settings           = ClusterSingletonManagerSettings(system)
+          )
+          val manager = system.actorOf(singletonProps, "EventLog-" + scope)
```
Comment (acjay) [inline src/main/scala/net/artsy/atomic/AtomicEventStore.scala]: Will this past tests using `context.actorOf`? If so, I think that would be the more standard approach. If not, we can figure it out later.

File: src/main/scala/net/artsy/atomic/AtomicEventStore.scala [hunk 4]
```diff
@@ -181,35 +186,66 @@ abstract class AtomicEventStore[EventType <: Serializable: Scoped: ClassTag, Val
...
+          val path = manager.path.toStringWithoutAddress
+
+          val proxyProps = ClusterSingletonProxy.props(
+            singletonManagerPath = "/user/EventLog-" + scope,
+            settings             = ClusterSingletonProxySettings(system)
+          )
+          val proxy = system.actorOf(
+            proxyProps //, name = s"$scope-proxy"
```
Comment (acjay) [inline src/main/scala/net/artsy/atomic/AtomicEventStore.scala]: We should either name this actor or drop the comment.

**Comment by acjay** [comment]
Nice! After all the work and research of getting this right, the actual solution feels reasonably straightforward and maintainable. Down the road, it may make sense for us to figure out how to trigger [proper restarts](http://doc.akka.io/docs/akka/current/general/supervision.html#What_Restarting_Means). Perhaps we can do that within the overall context of adding more rigorous fault tolerance to the system.

File: src/main/scala/net/artsy/atomic/AtomicEventStore.scala (hunk 3 above)
Comment (bhoggard) [inline src/main/scala/net/artsy/atomic/AtomicEventStore.scala]: No, it doesn't find the proxied actor and `ValidationRequest`s time out.

**Comment by acjay** [comment]
Awesome!! Good to go!

**Pull request merged**